lib/absinthe/subscription.ex

```diff
@@ -118,13 +119,14 @@ defmodule Absinthe.Subscription do
       }
     }
 
-    {:ok, _} = Registry.register(registry, field_key, doc_value)
-    {:ok, _} = Registry.register(registry, {self(), doc_id}, field_key)
+    _ = Registry.register(uniq_registry, {self(), :context}, doc.execution.context)
```

Comment:
I think this is going to have to match on `{:already_registered`, fetch the current context, and then have the value be some sort of map of `%{context_id => context}`.

This complicates unregister because it'll need to look at the counter and only remove the context when there are no documents remaining with that context_id. This may not be very efficient to lookup.

Reply:
Some general thoughts:

1. You don't need to store all information on the registry, only the information that you are going to read from another process. So for example, the map could be part of the process dictionary, or you could return a ref on subscribe which people need to use to unsubscribe (at least if they want it to be fast)

2.  Could you store `{self(), context_id}` instead of `:context`?

3. Maybe you don't need two registries. You can keep this as a duplicate registry and you do a read before you figure out if you need to store the context. On the other hand, lookups on duplicate registries are expensive, because you need to traverse all partitions.

Reply:
> On the other hand, lookups on duplicate registries are expensive, because you need to traverse all partitions.

I just pushed Registry.values/3 that allows you to read a key for a given pid, this makes looking up a key for a given pid efficient for both unique and duplicate registries.

There is also an argument to add a `has_key?` function but I am waiting a bit until we are sure we will need it.